### PR TITLE
In fixing funcunit to work for HTML5 drag and drop, we unfortunately …

### PR DESCRIPTION
**Pull request by kdillon**
…broke it for jQuery drag drop. The reason for this is that HTML5 stops sending mouse events and starts sending _drag_ events when a drag begins on a _draggable_ object. jQuery draggables do not necessarily have the _draggable_ attribute. So we now check for the _draggable_ attribute and send drag events if it is _draggable_, and if it isn't draggable we just send mousedown, mousemoves, and mouseups (etc).

One thing that I think I've done wrong here, is that the test requires jquery-ui.js . I wanted a "pure" jQuery drag and drop to test-against (rather than on draginit. I would like to cover all usecases of jQuery, not just jQuery++). And the examples that I found all used the technique that I used in the added testpage. Unfortunately that technique requires this external ui dependency.

I suspect that we would want this to be downloaded as part of node_modules, but I don't want to add new dependencies without the approval of the rest of the group.

Thanks!

### CODE REVIEW

**Comment by kdillon** [comment]
Sorry I meant to ask for Kevin Philips as a reviewer not an assignee.

**Comment by matthewp** [comment]
👍

File: test/tests.js [hunk 1]
```diff
@@ -5,6 +5,6 @@ require('test/mouse_move_test');
 require('test/mouse_button_test');
 //require('test/mouse_regressions_test');
 require('test/drag_test_basic');
-require('test/drag_test_offsets');
+//require('test/drag_test_offsets');
```
Comment (chasenlehara) [inline test/tests.js]: Are these tests not working anymore?

File: test/drag_test_basic.js [hunk 2]
```diff
@@ -29,8 +29,34 @@ QUnit.test("Drag Item Upward", 2, function () {
...
+		
+		// Timeout is required because jQuery isn't ready yet on the target page
+		setTimeout(function () {
+			syn.drag(draggable, {to: target}, function () {
+				var check = pageUnderTest.querySelector('#draggable')
+				ok(check.classList.contains('leftOfMark'), "jQuery drag did not register.");
+				
+				start();
+			});
+		}, 500);
```
Comment (chasenlehara) [inline test/drag_test_basic.js]: Hm, I thought the `load` event only fired when _everything_ had been loaded…

Comment (kdillon) [reply test/drag_test_basic.js]: I agree that its probably fired when all of the resources had loaded, but once it loads, the javascript must execute and alter the element in some way.

I don't want to give the wrong impression that I know how this works, js being single-threaded and all, but cloud the load event execution halt the jquery-ui wiring-up execution?

All I know is: with the pause, this works for me 10/10, without the pause it fails 10/10

File: test/tests.js (hunk 1 above)
Comment (kdillon) [reply test/tests.js]: These tests are for an upcoming feature that I am working on next.

Comment (chasenlehara) [reply test/tests.js]: Ah, this was added when you merged your pending work while making the last release, got it.

File: test/drag_test_basic.js (hunk 2 above)
Comment (chasenlehara) [reply test/drag_test_basic.js]: Yeah, I don’t think the load event should fire until after the jQuery script has executed in the frame, so maybe there’s something else that has to be waited on?

**Comment by chasenlehara** [comment]
When merging, I would use the “Squash and merge” option so the jQuery-UI file never gets committed into the repo.
screen shot 2017-10-31 at 12 42 36 pm

**Comment by kdillon** [comment]
Squash-merged and deleted the branch

**Pull request merged**